### ISSUE TITLE
show_debug_message asks for too many arguments.

### ISSUE THREAD

**Issue by catthou** [enigma-dev/enigma-dev#1044]
Bringing my Gamemaker Studio project over, the error I'm getting immediately is:

`Line 32, position 81 (absolute 933): Too few arguments to function 'show_debug_message': provided 1, required 2.`

However, both the yoyo and enigma wiki say show_debug_message() should only need 1 argument: the string to print. Ironically enough, the line throwing the error is:

`show_debug_message("csvToGrid: Error: Wrong number of arguments");`

...however commenting it out continues to throw the error on every other show_debug_message().

**Comment by JoshDreamland** [comment]
That's weird. What does your functions list (under the Compiler menu) show as the prototype of that function?

**Comment by catthou** [comment]
`show_debug_message : void show_debug_message(<NULL>, <NULL>);`
`show_message : int show_message(<NULL>, <NULL>);`

It seems show_message() also requires two arguments when it should only require one and is an int instead of void, so my list must be getting messed up somewhere. By far most of the functions in the function list look fine but more than a few are stuffed with NULLs, some shown here: http://imgur.com/ACpdagm - I'm not sure what it's _supposed_ to look like.

I'm on Linux Mint 18.1, and I installed Enigma by clicking on 'Code' above and selecting 'Clone or download' and choosing 'Download Zip', and then unzipping it to a clean directory and following the instructions in building.md to install necessary dependencies, then ran the python installer script. It opened LateralGM and built CompilerSource in a popup, with some warnings (mostly about indentation on your else statements :P) but no errors. Later I ran `make clean` and then `make` to manually build but that changed nothing.

I've been diddling around in LGM and haven't found anything else that seems broken. There are a ton of Globals but the Type list is empty. I don't know if that's intended or not?

**Comment by JoshDreamland** [comment]
I wrote ENIGMA's C++ parser before C++11 was finalized. Now that compilers are using C++14/17 features, it just can't accurately read it functions, anymore. I posted a workaround on the forum, where I have people define the macros that prevent the system string header from being included, then define a really simple string type on their own, mirroring the structure of the STL's. The workaround is [here​](http://enigma-dev.org/forums/index.php?topic=2669.msg25839#msg25839); give it a try.
